### ISSUE TITLE
KeyError: '1' when plotting MultiIndex data.

### ISSUE THREAD

**Issue by nistrup** [man-group/dtale#77]
First off all I absolutely low this package, thank you so much for your effort! I am however having some issues:

I have the following dataframe containing fundamental data (**columns**) for different stocks (**ticker**) at different times (**asof**):
![image](https://user-images.githubusercontent.com/17565925/75081673-4ee9bc00-5510-11ea-8ee1-e9846c152ad4.png)

So let's say I want to plot the evolution of a feature (**EQ_EARN_SHARE**) from Apple (**ticker == 'AAPL'**) over time (**asof**), I would suspect I should do this:
![image](https://user-images.githubusercontent.com/17565925/75081755-aab44500-5510-11ea-97db-8238cde7ff44.png)

Sadly this results in the error:
```
Traceback (most recent call last):
  File "C:\Users\Peter\Anaconda3\lib\site-packages\dtale\dash_application\charts.py", line 723, in build_figure_data
    data = run_query(DATA[data_id], query, CONTEXT_VARIABLES[data_id])
KeyError: '1'
```

What am I missing?

**Comment by aschonfeld** [comment]
My sincerest apologies, i think this is a breaking change in v1.7.6,  i’ll try pushing a fix tomorrow.  For the time being if you downgrade to 1.7.5 you should be fine

**Comment by nistrup** [comment]
@aschonfeld thank you so much, really love what you're doing, keep it up!

**Comment by aschonfeld** [comment]
Sorry I took a closer look at your chart configuration and if you wanted to plot EQ_EARN_SHARE over time then you'll want asof to be set to the X-Axis and EQ_EARN_SHARE to be set to the Y-Axis.  That is assuming the `KeyError` isn't still happening in 1.7.5

**Comment by aschonfeld** [comment]
Also, can you let me know what the order of operations was when bringing up D-Tale?  Did you open any sessions and then kill some then open some again?  I'm having a lot of trouble recreating the error.

Also can you let me know what version of python you're using as well as the versions of all the packages in your virtual environment?  You can get this by running either `conda list` or `pip freeze` depending on how you set up your environment.

Thanks

**Comment by aschonfeld** [comment]
Good news, I think I may have found what is causing this issue.  Hopefully will have a fix out there tomorrow.

FYI, I think v1.7.6 is fine to use.  This issue has been around a while, but its tough to hit. :(

**Comment by nistrup** [comment]
> Also, can you let me know what the order of operations was when bringing up D-Tale? Did you open any sessions and then kill some then open some again? I'm having a lot of trouble recreating the error.
> 
> Also can you let me know what version of python you're using as well as the versions of all the packages in your virtual environment? You can get this by running either `conda list` or `pip freeze` depending on how you set up your environment.
> 
> Thanks

```
# Name                    Version                   Build  Channel
_anaconda_depends         2019.03                  py37_0
_ipyw_jlab_nb_ext_conf    0.1.0                    py37_0
absl-py                   0.9.0                    py37_0
alabaster                 0.7.12                   py37_0
anaconda                  custom                   py37_1
anaconda-client           1.7.2                    py37_0
anaconda-navigator        1.9.7                    py37_0
anaconda-project          0.8.4                      py_0
asn1crypto                1.3.0                    py37_0
astor                     0.8.1                    pypi_0    pypi
astroid                   2.3.3                    py37_0
astropy                   4.0              py37he774522_0
atomicwrites              1.3.0                    py37_1
attr                      0.3.1                    pypi_0    pypi
attrs                     19.3.0                     py_0
autobahn                  19.11.1                  pypi_0    pypi
automat                   0.8.0                    pypi_0    pypi
autopep8                  1.4.4                      py_0
babel                     2.8.0                      py_0
backcall                  0.1.0                    py37_0
backports                 1.0                        py_2
backports.functools_lru_cache 1.6.1                      py_0
backports.os              0.1.1                    py37_0
backports.shutil_get_terminal_size 1.0.0                    py37_2
backports.tempfile        1.0                        py_1
backports.weakref         1.0.post1                  py_1
backtrader                1.9.74.123               pypi_0    pypi
beautifulsoup4            4.8.2                    py37_0
bitarray                  1.2.1            py37he774522_0
bitmex                    0.2.2                    pypi_0    pypi
bkcharts                  0.2                      py37_0
blas                      1.0                         mkl
bleach                    3.1.0                    py37_0
blinker                   1.4                      py37_0
blosc                     1.16.3               h7bd577a_0
bokeh                     1.4.0                    py37_0
boto                      2.49.0                   py37_0
bottleneck                1.3.1            py37h8c2d366_0
bravado                   10.4.2                   pypi_0    pypi
bravado-core              5.13.2                   pypi_0    pypi
bzip2                     1.0.8                he774522_0
ca-certificates           2020.1.1                      0
cachetools                4.0.0                    pypi_0    pypi
certifi                   2019.11.28               py37_0
cffi                      1.14.0           py37h7a1dbc1_0
chardet                   3.0.4                 py37_1003
click                     7.0                      py37_0
cloudpickle               1.3.0                      py_0
clyent                    1.2.2                    py37_1
colorama                  0.4.3                      py_0
comtypes                  1.1.7                    py37_0
conda                     4.8.2                    py37_0
conda-build               3.18.11                  py37_0
conda-env                 2.6.0                         1
conda-package-handling    1.6.0            py37h62dcd97_0
conda-verify              3.4.2                      py_1
console_shortcut          0.1.1                         3
constantly                15.1.0                   pypi_0    pypi
contextlib2               0.6.0.post1                py_0
cryptography              2.8              py37h7a1dbc1_0
curl                      7.68.0               h2a8f88b_0
cycler                    0.10.0                   py37_0
cython                    0.29.15          py37ha925a31_0
cytoolz                   0.10.1           py37he774522_0
dash                      1.9.0                    pypi_0    pypi
dash-core-components      1.8.0                    pypi_0    pypi
dash-daq                  0.3.3                    pypi_0    pypi
dash-html-components      1.0.2                    pypi_0    pypi
dash-renderer             1.2.4                    pypi_0    pypi
dash-table                4.6.0                    pypi_0    pypi
dask                      2.10.1                     py_0
dask-core                 2.10.1                     py_0
dateparser                0.7.2                    pypi_0    pypi
decorator                 4.4.1                      py_0
defusedxml                0.6.0                      py_0
distributed               2.10.0                     py_0
docutils                  0.16                     py37_0
dtale                     1.7.6                    pypi_0    pypi
entrypoints               0.3                      py37_0
et_xmlfile                1.0.1                    py37_0
fastcache                 1.1.0            py37he774522_0
filelock                  3.0.12                     py_0
flask                     1.1.1                      py_0
flask-compress            1.4.0                    pypi_0    pypi
freetype                  2.9.1                ha9979f8_1
fsspec                    0.6.2                      py_0
future                    0.18.2                   py37_0
gast                      0.3.3                      py_0
get_terminal_size         1.0.0                h38e98db_0
gevent                    1.4.0            py37he774522_0
glob2                     0.7                        py_0
google                    2.0.3                    pypi_0    pypi
google-auth               1.11.0                   pypi_0    pypi
google-auth-oauthlib      0.4.1                      py_2
google-pasta              0.1.8                    pypi_0    pypi
googletrans               2.4.0                    pypi_0    pypi
greenlet                  0.4.15           py37hfa6e2cd_0
grpcio                    1.27.1                   pypi_0    pypi
h5py                      2.10.0           py37h5e291fa_0
hdf5                      1.10.4               h7ebc959_0
heapdict                  1.0.1                      py_0
html5lib                  1.0.1                    py37_0
htmlmin                   0.1.12                   pypi_0    pypi
hyperlink                 19.0.0                   pypi_0    pypi
hypothesis                5.5.4                      py_0
icc_rt                    2019.0.0             h0cc432a_1
icu                       58.2                 ha66f8fd_1
idna                      2.8                      py37_0
imageio                   2.6.1                    py37_0
imagesize                 1.2.0                      py_0
importlib_metadata        1.5.0                    py37_0
incremental               17.5.0                   pypi_0    pypi
intel-openmp              2020.0                      166
ipykernel                 5.1.4            py37h39e3cac_0
ipython                   7.12.0           py37h5ca1d4c_0
ipython_genutils          0.2.0                    py37_0
ipywidgets                7.5.1                      py_0
isort                     4.3.21                   py37_0
itsdangerous              1.1.0                    py37_0
jdcal                     1.4.1                      py_0
jedi                      0.16.0                   py37_0
jinja2                    2.11.1                     py_0
joblib                    0.14.1                     py_0
jpeg                      9b                   hb83a4c4_2
json5                     0.9.1                      py_0
jsonpointer               2.0                      pypi_0    pypi
jsonref                   0.2                      pypi_0    pypi
jsonschema                3.2.0                    py37_0
jupyter                   1.0.0                    py37_7
jupyter_client            5.3.4                    py37_0
jupyter_console           6.1.0                      py_0
jupyter_contrib_core      0.3.3                      py_2    conda-forge
jupyter_contrib_nbextensions 0.5.1                    py37_0    conda-forge
jupyter_core              4.6.1                    py37_0
jupyter_highlight_selected_word 0.2.0                 py37_1000    conda-forge
jupyter_latex_envs        1.4.4                 py37_1000    conda-forge
jupyter_nbextensions_configurator 0.4.1                    py37_0    conda-forge
jupyterlab                1.2.6              pyhf63ae98_0
jupyterlab_server         1.0.6                      py_0
kaggle                    1.5.6                    pypi_0    pypi
keras-applications        1.0.8                    pypi_0    pypi
keras-preprocessing       1.1.0                    pypi_0    pypi
keyring                   21.1.0                   py37_0
kiwisolver                1.1.0            py37ha925a31_0
krb5                      1.17.1               hc04afaa_0
lazy-object-proxy         1.4.3            py37he774522_0
libarchive                3.3.3                h0643e63_5
libcurl                   7.68.0               h2a8f88b_0
libiconv                  1.15                 h1df5818_7
liblief                   0.9.0                ha925a31_2
libpng                    1.6.37               h2a8f88b_0
libprotobuf               3.11.4               h7bd577a_0
libsodium                 1.0.16               h9d3ae62_0
libssh2                   1.8.2                h7a1dbc1_0
libtiff                   4.1.0                h56a325e_0
libxml2                   2.9.9                h464c3ec_0
libxslt                   1.1.33               h579f668_0
llvmlite                  0.31.0           py37ha925a31_0
locket                    0.2.0                    py37_1
lxml                      4.5.0            py37h1350720_0
lz4                       3.0.2                    pypi_0    pypi
lz4-c                     1.8.1.2              h2fa13f4_0
lzo                       2.10                 h6df0209_2
m2w64-gcc-libgfortran     5.3.0                         6
m2w64-gcc-libs            5.3.0                         7
m2w64-gcc-libs-core       5.3.0                         7
m2w64-gmp                 6.1.0                         2
m2w64-libwinpthread-git   5.0.0.4634.697f757               2
markdown                  3.2                      pypi_0    pypi
markupsafe                1.1.1            py37he774522_0
matplotlib                3.1.3                    py37_0
matplotlib-base           3.1.3            py37h64f37c6_0
mccabe                    0.6.1                    py37_1
mechanize                 0.4.4                    pypi_0    pypi
menuinst                  1.4.16           py37he774522_0
mistune                   0.8.4            py37he774522_0
mkl                       2020.0                      166
mkl-service               2.3.0            py37hb782905_0
mkl_fft                   1.0.15           py37h14836fe_0
mkl_random                1.1.0            py37h675688f_0
mock                      4.0.1                      py_0
monotonic                 1.5                      pypi_0    pypi
more-itertools            8.2.0                      py_0
mpmath                    1.1.0                    py37_0
msgpack-python            0.5.6                    pypi_0    pypi
msys2-conda-epoch         20160418                      1
multipledispatch          0.6.0                    py37_0
navigator-updater         0.2.1                    py37_0
nbconvert                 5.6.1                    py37_0
nbformat                  5.0.4                      py_0
networkx                  2.4                        py_0
nltk                      3.4.5                    py37_0
nose                      1.3.7                    py37_2
notebook                  6.0.3                    py37_0
numba                     0.48.0           py37h47e9c7a_0
numexpr                   2.7.1            py37h25d0782_0
numpy                     1.18.1           py37h93ca92e_0
numpy-base                1.18.1           py37hc3f5095_1
numpydoc                  0.9.2                      py_0
oauthlib                  3.1.0                      py_0
olefile                   0.46                     py37_0
openpyxl                  3.0.3                      py_0
openssl                   1.1.1d               he774522_4
opt-einsum                3.1.0                    pypi_0    pypi
packaging                 20.1                       py_0
pandas                    0.25.3                   pypi_0    pypi
pandas-profiling          2.5.0                    pypi_0    pypi
pandoc                    2.2.3.2                       0
pandocfilters             1.4.2                    py37_1
parso                     0.6.1                      py_0
partd                     1.1.0                      py_0
path                      13.1.0                   py37_0
path.py                   12.4.0                        0
pathlib2                  2.3.5                    py37_0
patsy                     0.5.1                    py37_0
pep8                      1.7.1                    py37_0
phik                      0.9.9                    pypi_0    pypi
pickleshare               0.7.5                    py37_0
pillow                    7.0.0            py37hcc1f983_0
pip                       20.0.2                   py37_1
pivottablejs              0.9.0                    pypi_0    pypi
pkginfo                   1.5.0.1                  py37_0
plotly                    4.5.0                      py_0
pluggy                    0.13.1                   py37_0
ply                       3.11                     py37_0
powershell_shortcut       0.0.1                         2
prometheus_client         0.7.1                      py_0
prompt_toolkit            3.0.3                      py_0
protobuf                  3.11.3                   pypi_0    pypi
psutil                    5.6.7            py37he774522_0
py                        1.8.1                      py_0
py-lief                   0.9.0            py37ha925a31_2
pyasn1                    0.4.8                      py_0
pyasn1-modules            0.2.7                    pypi_0    pypi
pycodestyle               2.5.0                    py37_0
pycosat                   0.6.3            py37he774522_0
pycparser                 2.19                     py37_0
pycrypto                  2.6.1            py37hfa6e2cd_9
pycurl                    7.43.0.5         py37h7a1dbc1_0
pyflakes                  2.1.1                    py37_0
pygments                  2.5.2                      py_0
pyhamcrest                1.9.0                    pypi_0    pypi
pyjwt                     1.7.1                    py37_0
pylint                    2.4.4                    py37_0
pyodbc                    4.0.30           py37ha925a31_0
pyopenssl                 19.1.0                   py37_0
pyparsing                 2.4.6                      py_0
pyqt                      5.9.2            py37h6538335_2
pyreadline                2.1                      py37_1
pyrsistent                0.15.7           py37he774522_0
pysocks                   1.7.1                    py37_0
pytables                  3.6.1            py37h1da0976_0
pytest                    5.3.5                    py37_0
pytest-arraydiff          0.3              py37h39e3cac_0
pytest-astropy            0.8.0                      py_0
pytest-astropy-header     0.1.2                      py_0
pytest-doctestplus        0.5.0                      py_0
pytest-openfiles          0.4.0                      py_0
pytest-remotedata         0.3.2                    py37_0
python                    3.7.6                h60c2a47_2
python-binance            0.7.4                    pypi_0    pypi
python-dateutil           2.8.1                      py_0
python-libarchive-c       2.8                     py37_13
python-slugify            4.0.0                    pypi_0    pypi
pytz                      2019.3                     py_0
pywavelets                1.1.1            py37he774522_0
pywin32                   227              py37he774522_1
pywin32-ctypes            0.2.0                 py37_1000
pywinpty                  0.5.7                    py37_0
pyyaml                    5.3              py37he774522_0
pyzmq                     18.1.1           py37ha925a31_0
qt                        5.9.7            vc14h73c81de_0
qtawesome                 0.6.1                      py_0
qtconsole                 4.6.0                      py_1
qtpy                      1.9.0                      py_0
regex                     2019.12.20               pypi_0    pypi
requests                  2.23.0                   pypi_0    pypi
requests-oauthlib         1.3.0                      py_0
retrying                  1.3.3                    py37_2
rfc3987                   1.3.8                    pypi_0    pypi
rope                      0.16.0                     py_0
rsa                       4.0                        py_0
ruamel_yaml               0.15.87          py37he774522_0
scikit-image              0.16.2           py37h47e9c7a_0
scikit-learn              0.22.1           py37h6288b17_0
scipy                     1.4.1            py37h9439919_0
seaborn                   0.10.0                     py_0
selenium                  3.141.0                  pypi_0    pypi
send2trash                1.5.0                    py37_0
service-identity          18.1.0                   pypi_0    pypi
setuptools                45.2.0                   py37_0
simplegeneric             0.8.1                    py37_2
simplejson                3.16.0                   pypi_0    pypi
singledispatch            3.4.0.3                  py37_0
sip                       4.19.8           py37h6538335_0
six                       1.14.0                   py37_0
snappy                    1.1.7                h777316e_3
snowballstemmer           2.0.0                      py_0
sortedcollections         1.1.2                    py37_0
sortedcontainers          2.1.0                    py37_0
soupsieve                 1.9.5                    py37_0
sphinx                    2.4.0                      py_0
sphinxcontrib             1.0                      py37_1
sphinxcontrib-applehelp   1.0.1                      py_0
sphinxcontrib-devhelp     1.0.1                      py_0
sphinxcontrib-htmlhelp    1.0.2                      py_0
sphinxcontrib-jsmath      1.0.1                      py_0
sphinxcontrib-qthelp      1.0.2                      py_0
sphinxcontrib-serializinghtml 1.1.3                      py_0
sphinxcontrib-websupport  1.2.0                      py_0
spyder                    3.3.6                    py37_0
spyder-kernels            0.5.2                    py37_0
sqlalchemy                1.3.13           py37he774522_0
sqlite                    3.31.1               he774522_0
statsmodels               0.11.0           py37he774522_0
strict-rfc3339            0.7                      pypi_0    pypi
style                     1.1.0                    pypi_0    pypi
swagger-spec-validator    2.4.3                    pypi_0    pypi
sympy                     1.5.1                    py37_0
tangled-up-in-unicode     0.0.3                    pypi_0    pypi
tbb                       2020.0               h74a9793_0
tblib                     1.6.0                      py_0
tensorboard               2.1.0                     py3_0
tensorflow                2.1.0                    pypi_0    pypi
tensorflow-estimator      2.1.0                    pypi_0    pypi
termcolor                 1.1.0                    pypi_0    pypi
terminado                 0.8.3                    py37_0
testpath                  0.4.4                      py_0
text-unidecode            1.3                      pypi_0    pypi
tk                        8.6.8                hfa6e2cd_0
toolz                     0.10.0                     py_0
tornado                   6.0.3            py37he774522_3
tqdm                      4.43.0                   pypi_0    pypi
traitlets                 4.3.3                    py37_0
twisted                   19.10.0                  pypi_0    pypi
txaio                     18.8.1                   pypi_0    pypi
typed-ast                 1.4.0                    pypi_0    pypi
typing-extensions         3.7.4                    pypi_0    pypi
tzlocal                   2.0.0                    pypi_0    pypi
unicodecsv                0.14.1                   py37_0
update                    0.0.1                    pypi_0    pypi
urllib3                   1.24.3                   pypi_0    pypi
vc                        14.1                 h0510ff6_4
visions                   0.2.2                    pypi_0    pypi
vs2015_runtime            14.16.27012          hf0eaf9b_1
wcwidth                   0.1.8                      py_0
webcolors                 1.10                     pypi_0    pypi
webencodings              0.5.1                    py37_1
werkzeug                  1.0.0                      py_0
wheel                     0.34.2                   py37_0
widgetsnbextension        3.5.1                    py37_0
win_inet_pton             1.1.0                    py37_0
win_unicode_console       0.5                      py37_0
wincertstore              0.2                      py37_0
winpty                    0.4.3                         4
wrapt                     1.11.2           py37he774522_0
xlrd                      1.2.0                    py37_0
xlsxwriter                1.2.7                      py_0
xlwings                   0.17.1                   py37_0
xlwt                      1.3.0                    py37_0
xz                        5.2.4                h2fa13f4_4
yaml                      0.1.7                hc54c509_2
zeromq                    4.3.1                h33f27b4_3
zict                      1.0.0                      py_0
zipp                      2.2.0                      py_0
zlib                      1.2.11               h62dcd97_3
zope-interface            4.7.1                    pypi_0    pypi
zstd                      1.3.7                h508b16e_0
```

**Comment by nistrup** [comment]
> @nistrup Sorry I took a closer look at your chart configuration and if you wanted to plot EQ_EARN_SHARE over time then you'll want asof to be set to the X-Axis and EQ_EARN_SHARE to be set to the Y-Axis. That is assuming the `KeyError` isn't still happening in 1.7.5
> 
> Also, I'm able to produce a similar style chart on 1.7.6 using the demo site:
> http://andrewschonfeld.pythonanywhere.com/charts/1?chart_type=line&query=security_id+%3D%3D+100000&x=date&barmode=group&cpg=false&y=%5B%22Col0%22%5D
> 
> Is there any chance you can send the URL's you're seeing for the data grid and the charts page?

Yes you'r obviously correct regarding the x- and y-axis! I dosn't solve my problem though..

If I understand your question correctly, this is the URL I'm seeing for the charting: http://desktop-11np9kl:40000/charts/1

The first time I got this error is was actually `KeyError: '2'` and the URL for that error was ".../charts/2", I figure it was because I had already launched a d-tale "session"

**Comment by aschonfeld** [comment]
@nistrup thanks for getting me all this info.  I think I may have figured out what the problem is late last night. Its something with how i’ve structured the global variables so its a little more invasive than I thought but not tough, just need the kids to go to bed.  I’ll try releasing a new version tonight.

**Comment by adnanazad2065** [comment]
Hi @aschonfeld , I tried to replicate the charts following the exact dataset from your reddit demo, I got the same KeyError: '1'. Initially when I tried it on a large dataset got this as well, so likely not happening because of data size.


![image](https://user-images.githubusercontent.com/46439384/75094549-15c05300-55c7-11ea-8398-ae27f4cc2bf9.png)

**Comment by aschonfeld** [comment]
I got a little freed up and am close to pushing a new version that will hopefully solve this.  It should be noted that I was never able to truly replicate the issue, but I found something else that I think  has something to do with it. 🤞

**Comment by aschonfeld** [comment]
I've just made version 1.7.7 available on pypi if you want to try upgrading and see if that solves your issue.  Might be a little longer until its on conda.  Gotta wait for the build to run...

**Comment by nistrup** [comment]
> @nistrup @adnanazad2065 I've just made version 1.7.7 available on pypi if you want to try upgrading and see if that solves your issue. Might be a little longer until its on conda. Gotta wait for the build to run...

Thank you so much for your work! Good news is that the original error is gone, now I get this instead:

![image](https://user-images.githubusercontent.com/17565925/75097476-72ae1000-55ab-11ea-84a8-5e78598e860e.png)

**Comment by aschonfeld** [comment]
Arghh, that means that the data is still missing somehow.  Can you send me the order of operations you’re running to hit this? (Sorry to ask for yet more info). 

So are doing something like:
- restarting your kernel
- importing dtale
- calling dtale.show(df)
- going to the grid
- opening the charts page
- then running exactly what you showed in the screenshot?

Thanks again for your patience

**Comment by aschonfeld** [comment]
I think I know what the issue is.  As a way to guard against long running chart requests i wrapped the chart computation in a separate process with a timeout of 60 seconds.  It looks like for some reason your process cant access the global state.  I may have to drop this timeout because trying to pickle dataframes between prcoesses could be costly.

**Comment by nistrup** [comment]
> Arghh, that means that the data is still missing somehow. Can you send me the order of operations you’re running to hit this? (Sorry to ask for yet more info).
> 
> So are doing something like:
> 
> * restarting your kernel
> * importing dtale
> * calling dtale.show(df)
> * going to the grid
> * opening the charts page
> * then running exactly what you showed in the screenshot?
> 
> Thanks again for your patience

That's exactly right. So some information on my data:

DataFrame info:

```
<class 'pandas.core.frame.DataFrame'>
MultiIndex: 242467 entries, (A, 1999-11-30) to (ZTS, 2020-01-31)
Data columns (total 93 columns):
EQ_ADR_CHECK                         222883 non-null float64
EQ_ASSETS_SHARE                      222632 non-null float64
EQ_AVAIL_CHECK                       222883 non-null float64
EQ_BOOKVAL_SHARE                     225145 non-null float64
EQ_CAPEX_SHARE                       222632 non-null float64
...
...
...
EQ_UNFUNDEDPENSIONS_SHARE            222632 non-null float64
EQ_UNIVERSE                          217585 non-null float64
EQ_UP_REVISIONS                      224599 non-null float64
EQ_VOL_3M_LOC                        115947 non-null float64
EQ_WORKINGCAPITAL_SHARE              225145 non-null float64
dtypes: float64(93)
memory usage: 173.0+ MB
```

DataFrame sample:

![image](https://user-images.githubusercontent.com/17565925/75098497-3f717e00-55b7-11ea-83d1-a7258fa96175.png)

DataFrame in dtale:

![image](https://user-images.githubusercontent.com/17565925/75098506-692aa500-55b7-11ea-933e-450138089f4b.png)

The order of execution is quite like the one you exemplified:

1. imports
2. load and preprocess data
3. calling dtale.show(df)
4. going to the charts menu
5. putting in the exact values as the ones in my screenshot

**Comment by aschonfeld** [comment]
Its gotta be that subprocess.  Ok, to prove my theory you could try using version 1.7.0, although it may not be in conda so you’d have to go through pip.  Thats the only version with the plotly charts where I didnt use the timeout

**Comment by nistrup** [comment]
> Its gotta be that subprocess. Ok, to prove my theory you could try using version 1.7.0, although it may not be in conda so you’d have to go through pip. Thats the only version with the plotly charts where I didnt use the timeout

That fixed it!

![image](https://user-images.githubusercontent.com/17565925/75098770-8f9e0f80-55ba-11ea-859c-31f72262f2ae.png)

Thanks a lot, I'll use 1.7.0 for now then. Again thank you so much for your time, I greatly appreciate it!

**Comment by aschonfeld** [comment]
Thats a nice looking chart!

I’ll try taking out the multiprocessing tonight so you can get all the current functionality back.

The original reason for the multiprocessing was because if you request a really expensive computation then it blocks your entire D-Tale instance until it finishes and then you’ll have to create a new instance by either opening another notebook or calling dtale.show(df, port=40001, force=True)

FYI, you can use whatever port you want as long as its open :)

**Comment by aschonfeld** [comment]
So I have the new version ready to go but unfortunately CircleCI is undergoing maintenance for another hour and 45 minutes.  Hopefully I'll have it out soon after that.  Sorry

**Comment by aschonfeld** [comment]
I hope this isn't preemptive, but I'm going to close this. 🤞 I've just released 1.7.8 and it should hopefully solve your problem.  I've removed the multiprocessed timeouts.  It's available on pip and should be on conda soon.  Enjoy!
